src/lib/utils/device.utils.ts

```diff
@@ -56,18 +57,27 @@ export const isAndroidTablet = (): boolean => {
   return isAndroid() && !/mobile/i.test(a);
 };
 
+/**
+ * Current loophole (Feb 2024):
+ * - Firefox used on a desktop with touch screen will be identified as a mobile device.
+ */
 export const isMobile = (): boolean => {
   if (isNode()) {
     return false;
   }
 
-  const isTouchScreen: boolean = window.matchMedia(
-    "(any-pointer:coarse)",
-  ).matches;
-  const isMouseScreen: boolean =
-    window.matchMedia("(any-pointer:fine)").matches;
+  // Available in Chrome on any devices (Feb 2024)
+  // https://caniuse.com/mdn-api_navigator_useragentdata
+  if ("userAgentData" in navigator && nonNullish(navigator.userAgentData)) {
+    return navigator.userAgentData.mobile;
+  }
+
+  // Legacy
+  const isTouchScreen = window.matchMedia("(any-pointer:coarse)").matches;
```

Comment:
Let's hope there will be no macbooks w/ a touchscreen :D

Reply:
Indeed!

Worth to note that this "legacy" approach is similar to (not legacy) approach of [Ionic](https://github.com/ionic-team/ionic-framework/blob/35ab6b4816bd627239de8d8b25ce0c86db8c74b4/core/src/utils/platform.ts#L92).

Reply:
Ah, a good reference, thanks!